src/diff.c

```diff
@@ -607,103 +599,139 @@ int make_bsdiff_delta(char *old_filename, char *new_filename, char *delta_filena
 	eblen = 0;
 
 	/* Compute the differences */
-	scan = 0;
-	len = 0;
-	lastscan = 0;
-	lastpos = 0;
-	lastoffset = 0;
-	while (scan < new_size) {
-		oldscore = 0;
-
-		for (scsc = scan += len; scan < new_size; scan++) {
-			search(I, old_data, old_size, new_data + scan, new_size - scan,
-			       0, old_size, &pos, &len);
-
-			for (; scsc < scan + len; scsc++) {
-				if ((scsc + lastoffset < old_size) &&
-				    (old_data[scsc + lastoffset] == new_data[scsc])) {
-					oldscore++;
+	int64_t new_pos = 0;
+	int64_t old_pos = 0;
+	int64_t match_len = 0;
+	int64_t last_new_pos = 0;
+	int64_t last_old_pos = 0;
+	int64_t last_offset = 0;
+	while (new_pos < new_size) {
+		// Find an exact match between old and new files, and require
+		// that more than 8 of the matching bytes "mismatch" from the
+		// previous exact match. A score (old_score) is used to track
+		// how many bytes match starting from new_pos in new, and from
+		// old_pos in the previous iteration.
+		int64_t old_score = 0;
+		int64_t new_peek;
+		for (new_peek = new_pos += match_len; new_pos < new_size; new_pos++) {
```

Comment:
So new_peek gets the new_pos + match_len and new_pos gets the same thing. This is equivalent to 
```c
new_peek = new_pos = new_pos + match_len;
```
right? Just want to make sure I'm understanding this.

Reply:
oh my then it increments new_pos at the end...